fv3gfs-physics/fv3gfs/physics/stencils/update_dwind_phys.py

```diff
@@ -360,19 +380,19 @@ def __init__(self, stencil_factory: StencilFactory, grid, namelist, grid_info):
                         origin=origin_upper,
                         domain=self._domain_upper_north,
                     )
-            if self.grid.global_is < self._im2 and self._domain_lower_north[0] > 0:
+            if self.global_is < self._im2 and self._domain_lower_north[0] > 0:
                 self._copy3_stencil8 = stencil_factory.from_origin_domain(
                     copy3_stencil, origin=origin_lower, domain=self._domain_lower_north
                 )
         self._update_uwind_stencil = stencil_factory.from_origin_domain(
             update_uwind_stencil,
-            origin=(self.grid.halo, self.grid.halo, 0),
-            domain=(self.grid.nic, self.grid.njc + 1, self.grid.npz),
+            origin=(grid_indexing.n_halo, grid_indexing.n_halo, 0),
+            domain=(nic, njc + 1, npz),
         )
         self._update_vwind_stencil = stencil_factory.from_origin_domain(
             update_vwind_stencil,
-            origin=(self.grid.halo, self.grid.halo, 0),
-            domain=(self.grid.nic + 1, self.grid.njc, self.grid.npz),
+            origin=(grid_indexing.n_halo, grid_indexing.n_halo, 0),
```

Comment:
So much indexing logic we have to do here! At least it gets cached since it's done on init.